core/src/main/java/com/linecorp/armeria/server/throttling/FileBasedDynamicRequestThrottlingMapping.java

```diff
@@ -0,0 +1,132 @@
+        Thread thread = new Thread(() -> {
+            try {
+                while (true) {
+                    try {
+                        WatchKey key = watchService.take();
+                        for (WatchEvent<?> event : key.pollEvents()) {
+                            // We only register "ENTRY_MODIFY" so the context is always a Path.
+                            final Path changed = (Path) event.context();
+                            if (changed.equals(settingFile.getFileName())) {
+                                updateThrottlingSettings();
+                                break;
+                            }
+                        }
+                        if (!key.reset()) {
+                            break;
+                        }
+                    } catch (InterruptedException e) {
+                        throw new IllegalStateException("Interrupted while taking from file.", e);
+                    }
+
+                }
+            } catch (ClosedWatchServiceException ignored) {
+                logger.info("Stopping hosts file watch thread.");
+            }
+        });
+        thread.setDaemon(true);
+        thread.start();
+        return thread;
+    }
+
+    private void updateThrottlingSettings() {
+        Map<String, String> newStrategies;
+        try {
+            newStrategies = OBJECT_MAPPER.readValue(settingFile.toFile(),
+                                                    new TypeReference<Map<String, String>>() {});
+        } catch (IOException e) {
+            logger.warn("Could not read hosts file", e);
+            return;
+        }
+
+        strategies = newStrategies.entrySet().stream()
+                                  .collect(toImmutableMap(Entry::getKey,
+                                                          e -> convertToThrottlingStrategy(e.getValue())));
+    }
+
+    private static RequestThrottlingStrategy convertToThrottlingStrategy(String throttlingSpec) {
+        List<String> spec = COLON_SPLITTER.splitToList(throttlingSpec);
+        if (spec.size() != 2) {
+            logger.warn("Invalid throttlingSpec: {}", throttlingSpec);
+            return RequestThrottlingStrategy.never();
+        }
+        switch (Ascii.toLowerCase(spec.get(0))) {
+            case "always":
+                return RequestThrottlingStrategy.always();
+            case "never":
+                return RequestThrottlingStrategy.never();
+            case "rate":
+                float rate = Floats.tryParse(spec.get(1));
+                return req -> CompletableFuture.completedFuture(rate > ThreadLocalRandom.current().nextFloat());
+            default:
+                throw new IllegalArgumentException("Unknown key " + spec.get(0));
+        }
+    }
+
+    @Override
+    public RequestThrottlingStrategy get(ServiceRequestContext ctx, Request req) {
+        String method = req instanceof RpcRequest ? ((RpcRequest) req).method() : ctx.method().name();
```

Comment:
Can we do like path#method ? sometimes same method name may bind to different path. And I think it's good to using same pattern for http handler. e.g. `/api#POST`.